src/main/java/common/NoGoodException.java

```diff
@@ -0,0 +1,8 @@
+package common;
+
+/** An exception to be thrown when the user input is not valid.  */
+public class NoGoodException extends Exception {
```

Comment:
At the first sight, I am not sure what does NoGoodException means, according to javaDoc, perhaps you could name it InvalidInputException